src/lexer_tok.c

```diff
@@ -107,16 +101,13 @@ t_error	ft_partition(t_src *src, t_buf *tmp)
 			ft_add_quoted_str(c, src, tmp);
 		else if ((c == ' ' || c == '\t') && tmp->cur_pos > 0)
 			break ;
-		else if ((c == '\n' || c == '|') && tmp->cur_pos > 0)
-		{
-			ft_unget_char(src);
-			break ;
-		}
+		else if (ft_strchr("\n|<>", c) && tmp->cur_pos > 0)
+			return (ft_unget_char(src));
 		else
 		{
 			ft_add_to_buf(c, tmp);
-			if (c == '\n' || c == '|')
-				break ;
+			if (ft_strchr("\n|<>", c))
```

Comment:
Maybe I'm missing something but I don't think we need to check for newline and pipe?
If it's just two characters I prefer the explicit way of checking (i.e. c == '<') since it's easier to read. It also needs no function call.

Reply:
We still have to check for \n and | to break out of loop